base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs

```diff
@@ -214,6 +214,13 @@ impl SenderTransactionInitializer {
     ) -> Result<&mut Self, BuildError> {
         let commitment_factory = PedersenCommitmentFactory::default();
         let commitment = commitment_factory.commit(&output.spending_key, &PrivateKey::from(output.value));
+        let recovery_byte = OutputFeatures::create_unique_recovery_byte(&commitment, self.rewind_data.as_ref());
+        if recovery_byte != output.features.recovery_byte {
+            self.clone().build_err(&*format!(
```

Comment:
This clone feels wrong. Wouldn't this error get lost?

Reply:
Without the clone: `cannot move out of `*self` which is behind a mutable reference`